verification/builder_test.go

```diff
@@ -166,19 +226,118 @@ func Test_VerifyWorkflowIdentity(t *testing.T) {
 			},
 			source: "github.com/asraa/slsa-on-github-test",
 		},
+		{
+			name: "valid workflow identity with fully qualified source - match builderID",
+			workflow: &WorkflowIdentity{
+				CallerRepository:  "asraa/slsa-on-github-test",
+				CallerHash:        "0dfcd24824432c4ce587f79c918eef8fc2c44d7b",
+				JobWobWorkflowRef: trustedBuilderRepository + "/.github/workflows/builder_go_slsa3.yml@refs/tags/v1.2.3",
+				Trigger:           "workflow_dispatch",
+				Issuer:            certOidcIssuer,
+			},
+			source: "github.com/asraa/slsa-on-github-test",
+			buildOpts: &BuilderOpts{
+				ExpectedID: asStringPointer("https://github.com/" + trustedBuilderRepository + "/.github/workflows/builder_go_slsa3.yml"),
+			},
+		},
+		{
+			name: "valid workflow identity with fully qualified source - mismatch builderID",
+			workflow: &WorkflowIdentity{
+				CallerRepository:  "asraa/slsa-on-github-test",
+				CallerHash:        "0dfcd24824432c4ce587f79c918eef8fc2c44d7b",
+				JobWobWorkflowRef: trustedBuilderRepository + "/.github/workflows/builder_go_slsa3.yml@refs/tags/v1.2.3",
+				Trigger:           "workflow_dispatch",
+				Issuer:            certOidcIssuer,
+			},
+			source: "github.com/asraa/slsa-on-github-test",
+			buildOpts: &BuilderOpts{
+				ExpectedID: asStringPointer("some-other-builderID"),
+			},
+			err: ErrorUntrustedReusableWorkflow,
+		},
 	}
 	for _, tt := range tests {
 		tt := tt // Re-initializing variable so it is not changed while executing the closure below
 		t.Run(tt.name, func(t *testing.T) {
 			t.Parallel()
-			err := VerifyWorkflowIdentity(tt.workflow, tt.source)
+			opts := tt.buildOpts
+			if opts == nil {
+				opts = &BuilderOpts{}
+			}
+			_, err := VerifyWorkflowIdentity(tt.workflow, opts, tt.source)
```

Comment:
nit: Consider allowing `VerifyWorkflowIdentity` to take a nil `BuilderOpts`

Reply:
I'll keep this way for now since each field will be a pointer. I'll keep this in mind for future iteration. Let me know if that's OK.

Reply:
Yeah, it's minor but I just mean basically putting the if above inside the function and allow callers to do `VerifyWorkflowIdentity(workflow, nil, source)`

Reply:
I'll do that in the #187 PR, thanks